mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/function/FunctionExecutorFactory.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.function;
+
+import org.mule.runtime.api.el.ExpressionFunction;
+import org.mule.runtime.api.meta.model.function.FunctionModel;
+import org.mule.runtime.extension.api.runtime.operation.OperationExecutor;
+
+public interface FunctionExecutorFactory {
+
+  /**
+   * Creates a new {@link ExpressionFunction}
+   *
+   * @param functionModel the model of the function to be executed
+   * @return a new {@link OperationExecutor}
+   */
+  ExpressionFunction createExecutor(FunctionModel functionModel, DefaultValueResolverFactory defaultResolverFactory);
```

Comment:
This should return a FunctionExecutor, so that the ExpressionFunction is encapsulated. This will not only make it more open to change, it will be easier to execute custom stuff around the function's call method (otherwise, you have to create an ExpressionFunction wrapper which is harder to maintain).

Plus, if the class is called FunctionExecutorFactory, it should return a FunctionExecutor